File: api/src/main/java/me/kvdpxne/dtm/arena/ArenaException.java
package me.kvdpxne.dtm.arena;

import me.kvdpxne.dtm.CodecException;

public class ArenaException extends CodecException {

  public ArenaException(String message, String code) {
    super(message, code);
  }

  public ArenaException(String message, Throwable cause, String code) {
    super(message, cause, code);
  }
}

File: src/main/kotlin/me/kvdpxne/dtm/arena/ArenaException.kt
package me.kvdpxne.dtm.arena

/**
 * Custom exception class for handling errors related to arenas in the game.
 *
 * This class extends [RuntimeException] to provide a specific exception type
 * for arena-related operations.
 *
 * @param message The detail message for the exception, which can provide
 *                information about the error.
 *
 * @since 0.1.0
 */
open class ArenaException(
  // @formatter:off
  message: String     = "",
  cause  : Throwable? = null
  // @formatter:on
) : RuntimeException(message, cause) {

  companion object {

    /**
     * Serial version UID for serialization compatibility.
     *
     * @since 0.1.0
     */
    @Suppress("ConstPropertyName")
    private const val serialVersionUID: Long = -8396776841121661460L
  }
}